convert_train_to_lmdb.py

```diff
@@ -0,0 +1,170 @@
+            digits[idx] = int(label_of_digit if label_of_digit != 10 else 0)    # label 10 is essentially digit zero
+
+        attrs_left, attrs_top, attrs_width, attrs_height = map(lambda x: [int(i) for i in x], [attrs['left'], attrs['top'], attrs['width'], attrs['height']])
+        min_left, min_top, max_right, max_bottom = (min(attrs_left),
+                                                    min(attrs_top),
+                                                    max(map(lambda x, y: x + y, attrs_left, attrs_width)),
+                                                    max(map(lambda x, y: x + y, attrs_top, attrs_height)))
+
+        center_x, center_y, max_side = ((min_left + max_right) / 2.0,
+                                        (min_top + max_bottom) / 2.0,
+                                        max(max_right - min_left, max_bottom - min_top))
+        bbox_left, bbox_top, bbox_width, bbox_height = (center_x - max_side / 2.0,
+                                                        center_y - max_side / 2.0,
+                                                        max_side,
+                                                        max_side)
+        image = np.array(ExampleReader._preprocess(Image.open(path_to_image_file), bbox_left, bbox_top, bbox_width, bbox_height)).tobytes()
+
+        example = example_pb2.Example()
+        example.image = image
+        example.length = length
+        example.digits.extend(digits)
+        return example
+
+
+def convert_to_lmdb(path_to_dataset_dir_and_digit_struct_mat_file_tuples,
+                    path_to_lmdb_dirs, choose_writer_callback):
+    num_examples = []
+    writers = []
+
+    for path_to_lmdb_dir in path_to_lmdb_dirs:
+        num_examples.append(0)
+        writers.append(lmdb.open(path_to_lmdb_dir, map_size=10*1024*1024*1024))
+
+    for path_to_dataset_dir, path_to_digit_struct_mat_file in path_to_dataset_dir_and_digit_struct_mat_file_tuples:
+        path_to_image_files = glob.glob(os.path.join(path_to_dataset_dir, '*.png'))
+        total_files = len(path_to_image_files)
+        print('%d files found in %s' % (total_files, path_to_dataset_dir))
+
+        with h5py.File(path_to_digit_struct_mat_file, 'r') as digit_struct_mat_file:
+            example_reader = ExampleReader(path_to_image_files)
+            block_size = 10000
+
+            for i in range(0, total_files, block_size):
+                txns = [writer.begin(write=True) for writer in writers]
+
+                for offset in range(block_size):
+                    idx = choose_writer_callback(path_to_lmdb_dirs)
+                    txn = txns[idx]
+
+                    example = example_reader.read_and_convert(digit_struct_mat_file)
+                    if example is None:
+                        break
+
+                    str_id = '{:08}'.format(num_examples[idx] + 1)
+                    txn.put(str_id.encode(), example.SerializeToString())
+                    num_examples[idx] += 1
```

Comment:
I made some changes to the original `convert_to_lmdb` around here for getting the image names, if u wanna also include that

Reply:
They're in the other branch `add-validation-losses`, the changes will be saved when we merge

Reply:
It was in a different file so it won't be applied here.

Reply:
oh right, I'm no longer using this file anyway so just leave it be for now